gutterball/src/main/java/org/candlepin/gutterball/report/ConsumerStatusReport.java

```diff
@@ -151,14 +173,60 @@ protected ReportResult execute(MultivaluedMap<String, String> queryParams, PageR
             queryParams.getFirst(CUSTOM_RESULTS_PARAM) : "";
         boolean useCustom = PropertyConverter.toBoolean(custom);
 
-        Page<Iterator<Compliance>> page = this.complianceSnapshotCurator.getSnapshotIterator(
-            targetDate,
-            consumerIds,
-            ownerFilters,
-            statusFilters,
-            attributeFilters,
-            pageRequest
-        );
+        Page<Iterator<Compliance>> page;
+
+        if (productNameFilters != null) {
+            page = this.complianceSnapshotCurator.getSnapshotIterator(
+                targetDate,
+                consumerIds,
+                ownerFilters,
+                statusFilters,
+                productNameFilters,
+                null,
+                null,
+                null,
+                pageRequest
+            );
+        }
+        else if (subscriptionSkuFilters != null) {
+            page = this.complianceSnapshotCurator.getSnapshotIterator(
+                targetDate,
+                consumerIds,
+                ownerFilters,
+                statusFilters,
+                null,
+                subscriptionSkuFilters,
+                null,
+                null,
+                pageRequest
+            );
+        }
+        else if (subscriptionNameFilters != null) {
+            page = this.complianceSnapshotCurator.getSnapshotIterator(
+                targetDate,
+                consumerIds,
+                ownerFilters,
+                statusFilters,
+                null,
+                null,
+                subscriptionNameFilters,
+                null,
+                pageRequest
+            );
+        }
+        else {
+            page = this.complianceSnapshotCurator.getSnapshotIterator(
+                targetDate,
+                consumerIds,
+                ownerFilters,
+                statusFilters,
+                null,
+                null,
+                null,
+                attributeFilters,
+                pageRequest
+            );
+        }
 
```

Comment:
Would it make sense to create specific methods in ComplianceSnapshotCurator for the calls to getSnapshotIterator above? Would probably be easier to understand what the method is doing. Looks like passing 'null' for certain parameters has a reason... perhaps we can make that more aparent.